### PR TITLE
[Maintenance] Add an exception for unavailable storages

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A
| --------------- | -----
| Bug fix?        | kinda
| New feature?    | yes?
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

There's a bunch of issues open Sylius regarding cli/stateless actions resulting in fatals since session is not available when resolving something within a context.
Would be nice to have a unified way of knowing whether the storage medium can be accessed at all.

### CODE REVIEW

File: src/Bundle/Storage/SessionStorage.php
Comment (diimpp) [inline src/Bundle/Storage/SessionStorage.php]: Since there is ongoing initiative to move namespaces in 1.11, new exception should go directly into target namespace.
 
```suggestion
use Sylius\Resource\Exception\StorageUnavailableException;
```

File: src/Bundle/spec/Storage/SessionStorageSpec.php
Comment (diimpp) [inline src/Bundle/spec/Storage/SessionStorageSpec.php]:
```suggestion
use Sylius\Resource\Exception\StorageUnavailableException;
```

File: src/Component/legacy/src/Exception/StorageUnavailableException.php
Comment (diimpp) [inline src/Component/legacy/src/Exception/StorageUnavailableException.php]: If for some reason legacy exception needs to be created, then it should become class alias, otherwise please move it to `src/Component/src/Exception/StorageUnavailableException`

File: src/Component/legacy/src/Exception/StorageUnavailableException.php
Comment (diimpp) [inline src/Component/legacy/src/Exception/StorageUnavailableException.php]: Exceptions are modernized at new non-component namespace, please apply this change.

```suggestion
use Sylius\Resource\Exception\RuntimeException;

class StorageUnavailableException extends RuntimeException
```

image

File: src/Component/src/Storage/StorageInterface.php
Comment (diimpp) [inline src/Component/src/Storage/StorageInterface.php]:
```suggestion
use Sylius\Resource\Exception\StorageUnavailableException;
```

**Review by diimpp — DISMISSED** [review]

**Comment by NoResponseMate** [comment]
@diimpp
Wanted to wait until #768 gets merged to avoid unnecessary conflicts.
Updated 👋

**Review by loic425 — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @NoResponseMate!

**Pull request merged**
